### ISSUE TITLE
eas-cli@11 breaks environment variable resolution

### ISSUE THREAD

**Issue by djMax** [expo/eas-cli#2519]
### Summary

I have an eas.json like so:
```
{
  "cli": {
    "version": ">= 7.0.0"
  },
  "build": {
    "development": {
      "developmentClient": true,
      "channel": "development",
      "env": {
        "EXPO_PUBLIC_APP_TYPE": "flavora",
        "EXPO_PUBLIC_APP_ENV": "development",
      }
    },
```

and so on. In v10, these environment variables are correctly given to app.config.ts to do its job of building a dynamic environment aware config. In version 11, this has stopped working an no environment variables are defined, breaking generation of app.config.

### Managed or bare?

managed

### Environment

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.6.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.12.2 - ~/.local/share/mise/installs/node/20/bin/node
      Yarn: 3.8.1 - ~/.local/share/mise/installs/node/20/bin/yarn
      npm: 10.5.0 - ~/.local/share/mise/installs/node/20/bin/npm
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2023.2 AI-232.10300.40.2321.11668458
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 11.0.0
    Expo Workflow: managed
```

### Error output

Project config: Slug for project identified by "extra.eas.projectId" (myapp-flavora) does not match the "slug" field (myapp-undefined). Learn more

### Reproducible demo or steps to reproduce from a blank project

I do not yet have one. This is a pretty major undocumented breaking change, so would love to either get a rationale or place to look first.

**Comment by itajenglish** [comment]
+1 Experiencing the same issue

**Comment by IjzerenHein** [comment]
Same here, we are trying to access the env-vars that are set by `eas-cli` in `app.config.ts` but it's no longer possible as of version 11

**Comment by szdziedzic** [comment]
Thanks for the report!

I was able to replicate the issue. This was indeed an unintended change, sorry about this. We added a static app config context definition to the `eas build` command in https://github.com/expo/eas-cli/commit/56510f092527e458aa9dca19feac0b5d425dac56#diff-cd2e00f4d286a4714f8736e14c635f4c3b98c402367cd71a83e4cd743034bd79, it caused `eas build` to try to resolve the project ID and slug from dynamic `app.config.ts` file without using env vars, which was the root cause of the issue in your case.

https://github.com/expo/eas-cli/pull/2521 should fix it.

It seems like the issue affected projects with dynamic slug/projectId fields in `app.config.ts`
